### ISSUE TITLE
Image Generation resulting in black images

### ISSUE THREAD

**Issue by MatheusHiro** [huggingface/diffusers#6492]
### Describe the bug

I've trained a unconditional model with a set of images and when sampling I am getting a black image. I've also tried setting the torch_dtype to float32 when sampling, but it didn't work

### Reproduction

**Training script:**
accelerate launch train_unconditional.py \
  --dataset_name="MatheusHiro/Cells" \
  --output_dir="ddpm-ema-cells-128" \
  --mixed_precision="fp16" \
  --resolution="128"

**Sampling:** 
from diffusers import DiffusionPipeline
import torch

pipeline = DiffusionPipeline.from_pretrained("./ddpm-ema-cells-128").to("cuda")
image = pipeline().images[0]
image.save("generated_image.png")

### Logs

```shell
(.env) (base) matheus@maxwell:/hdd/diffusers/diffusers/examples/unconditional_image_generation$ python sample.py 
Loading pipeline components...: 100%|█████████████| 2/2 [00:00<00:00,  2.22it/s]
100%|███████████████████████████████████████| 1000/1000 [00:39<00:00, 25.28it/s]
/hdd/diffusers/.env/lib/python3.10/site-packages/diffusers/utils/pil_utils.py:45: RuntimeWarning: invalid value encountered in cast
  images = (images * 255).round().astype("uint8")
```

### System Info

- GeForce GTX TITAN X 
- diffusers version: 0.25.0.dev0
- Python version: Python 3.10.9
- PyTorch version (GPU?): 2.1.2+cu12 (True)
- Huggingface_hub version: 0.19.4
- Transformers version: 4.36.1
- Accelerate version: 0.25.0
- xFormers version: 0.0.23.post1

**Comment by JeanKaddour** [comment]
I just came here to say that I experienced the same bug when running the [SDXL Fine-tuning example](https://github.com/huggingface/diffusers/blob/main/examples/text_to_image/README_sdxl.md).

To understand whether my training may cause this, I slightly modified the script to run the image generation with the validation prompt *before* the training starts and still experienced the same issue (plain black images logged in wandb).

Launch command can be found [here](https://github.com/huggingface/diffusers/blob/main/examples/text_to_image/README_sdxl.md#training)

**Comment by DN6** [comment]
Hi @MatheusHiro could you try the following?

1. Run a forward pass through the UNet and verify that there are no NaNs in the output tensor
2. Decode the output tensor from the previous step with the VAE and check if there are NaNs in the output tensor. 

This helps us identify which component might be causing the issue.

**Comment by Tauseefahmed1451** [comment]
I ran 
```
accelerate launch train_controlnet_sdxl.py \
 --pretrained_model_name_or_path=$MODEL_DIR \
 --output_dir=$OUTPUT_DIR \
 --dataset_name=xxxxxxxxx \
 --mixed_precision="fp16" \
 --resolution=1024 \
 --learning_rate=1e-5 \
 --max_train_steps=8000 \
--checkpointing_steps=4000 \
 --validation_image "./224.png" "./225.png" \
 --validation_prompt "xxxxxxx" \
 --validation_steps=5 \
--num_validation_images=1\
 --train_batch_size=1 \
 --report_to="wandb" 
```
with num_validation_images= 4/1 etc, changed learning rate, with and without prompts, but the images are returning black on wandb. I am using A6000 over on trainML. I ran the same script for sd1.5 and sd2.1 on colab and the images were not black. 


xxxxx = something I cannot disclose.
